### ISSUE TITLE
Aggregated contig statistics per taxon in draw_heatmaps

### ISSUE THREAD

**Issue by samnooij** [DennisSchmitz/Jovian_archive#37]
As also suggested in issue #12, the heatmaps can be more informative and smaller in (file) size if the information is aggregated on a taxon level. I.e. sometimes more than one contig matches a taxon, in which case the mouse-over information box will show several panels. This becomes problematic when there are more than 3 (or so) contigs: some panels might be drawn outside the figure box, invisible to the user. Moreover, with so many panels to load, it may take a while before the information box pops up. Even just opening the file may take long (if it opens at all).

Therefore, it would be much better to aggregate all relevant information to make only one panel for each heatmap block/taxon.

The information I suggest to insert is:
 1. Sample name
 2. Number of contigs
 3. Taxon name as final assignment by Jovian (most common or random pick in case of multiple?)
 4. Taxon (at the rank of the heatmap - makes it easier to read with very large heatmaps)
 5. Number of reads mapped to contig(s)
 6. Contig length(s; e.g. minimum and maximum)
 7. Average depth of coverage

Please leave a comment if you think other information should be there!

**Comment by samnooij** [comment]
This issue should be handled before #39: otherwise the files will probably become too big to open!

**Comment by samnooij** [comment]
_More technical details/note to self:_

By now I may also add the "number of ORFs" field. I think the best update for now would be to summarise all statistics at the beginning of `draw_heatmaps(...)` (e.g. insert a `groupby` line and create new dataframes/bokeh datasource at line 337 - or a new `group_by_rank(df = ..., rank = ...)` function).  
These data should include all categories listed above, and probably the statistics _minimum, average, and maximum_ provide the most convenient numbers.

_edit: note that the downside would be that even fields with 1 contigs would then show min, avg, max fields. I might check before grouping if there are duplicates in the current rank. If not, it should be possible to make heatmaps like they are now._

**Comment by DennisSchmitz** [comment]
@samnooij & @thierryjanssens , Thierry, weren't you also working on the ORFs code for the protein annotation? I might be confusing things, but if so, it would be good if you first discuss the changes and streamline it.

**Comment by samnooij** [comment]
I have pushed an update in branch 'heatmap_updates' (see commit https://github.com/DennisSchmitz/Jovian/commit/801c4a784152e5ed2c3006e8d3ac2e0380ad7957 and the three before it) that automatically detects cases in which samples have many contigs of the same taxonomic classifications. (I made the cutoff > 3 at the moment.) These would otherwise create enormous 'hover tooltips' (those information panels that you get when you mouse-over a block), but now they are aggregated in one, and the tooltip changes correspondingly. 'Smaller' heatmaps remain the same.

In my little test, file sizes are up to 70 kB, whereas otherwise they would be up to 1.4 MB.

And by the way, I have also included the number of open reading frames (ORFs) in the tooltip now. @DennisSchmitz : If you agree this is an improvement, I can continue and solve #39.

_edit: corrected file sizes_
